src/style-spec/expression/evaluation_context.js

```diff
@@ -56,10 +53,18 @@ class EvaluationContext {
 
     distanceFromCamera() {
         if (this.cameraDistanceReferencePoint && this.globals && this.globals.cameraDistanceMatrix) {
-            tempArray[0] = this.cameraDistanceReferencePoint.x;
-            tempArray[1] = this.cameraDistanceReferencePoint.y;
-            tempArray[2] = this.cameraDistanceReferencePoint.z;
-            return vec3.length(vec3.transformMat4(tempArray, tempArray, this.globals.cameraDistanceMatrix));
+
+            const m = this.globals.cameraDistanceMatrix;
+            const {x, y, z} = this.cameraDistanceReferencePoint;
+
+            //inlined vec3*mat4 multiplication to prevent importing gl-matrix as a dependency
```

Comment:
What's your reason to preferring in-lining to the dependency?

Reply:
Mostly so as to not bloat the `style-spec` package unnecessarily. It doesn't need matrix math anywhere else so `gl-matrix` isn't a dependency yet, and adding it just for this one new function doesn't feel like its necessary.